qlib/data/dataset/__init__.py

```diff
@@ -358,7 +363,6 @@ def __init__(
         )
         self.idx_arr = np.array(self.idx_df.values, dtype=np.float64)  # for better performance
 
-        del self.data  # save memory
```

Comment:
I think it will be better to delete this attribute even after you call self.data.drop.
Less attributes will make the interface simpler.